WordPress/gutenberg
# refactor: Improve useIsConnected Hook

## What?

Improve the recently merged `useIsConnected` Hook to align with project
practices and cover with automated tests.

## Why?

Fixes https://github.com/wordpress-mobile/gutenberg-mobile/issues/6472. Improve approachability and stability of the Hook.

## How?

- Rename `useIsConnected` to `useNetworkConnectivity`
- Add `useNetworkConnectivity` Hook tests
- Document `useNetworkConnectivity` Hook
- Rename `withIsConnected` to `withNetworkConnectivity`
- Optimistically presume network connectivity

## Testing Instructions




1. Clone the branch.
1. Launch the editor.
1. Disable the network connection (e.g. enable airplane mode).
1. Verify the offline status notice is displayed.
1. Enable the network connection (e.g. disable airplane mode).
1. Verify the offline status notice is not displayed.

### Testing Instructions for Keyboard

N/A, no UI changes included.

## Screenshots or screencast 
N/A, no UI changes included.
